wandb/sdk/artifacts/artifact.py

```
@@ -1686,6 +1686,78 @@ def download(
+                settings=wandb.Settings(silent="true"),
+            ):
+                return FilePathStr(
+                    self._run_artifact_download(
+                        root=root,
+                        recursive=recursive,
+                        allow_missing_references=allow_missing_references,
+                    )
+                    or ""
+                )
+        else:
+            return FilePathStr(
+                self._run_artifact_download(
+                    root=root,
+                    recursive=recursive,
+                    allow_missing_references=allow_missing_references,
+                )
+                or ""
+            )
+
+    def _run_artifact_download(
+        self,
+        root: Optional[str] = None,
+        recursive: bool = False,
+        allow_missing_references: bool = False,
+    ) -> Optional[FilePathStr]:
+        assert wandb.run is not None, "failed to initialize run"
+        run = wandb.run
+        if run._settings._require_nexus:
+            if run._backend and run._backend.interface:
+                if not run._settings._offline:
+                    result = run._backend.interface.communicate_download_artifact(
+                        self.qualified_name,
+                        root,
+                        recursive,
+                        allow_missing_references,
+                    )
+                    if result is not None:
+                        if result.response.download_artifact_response.error_message:
+                            raise ValueError(
+                                f"Error downloading artifact: {result.response.download_artifact_response.error_message}"
+                            )
+                        download_path = (
+                            result.response.download_artifact_response.file_download_path
+                        )
+                        return FilePathStr(download_path)
+                    return FilePathStr("")
```

Let's raise an exception in this case instead of returning an invalid value.

---

raised a `RuntimeError`